sdk/dapp-kit/src/components/AutoConnectWallet.tsx

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import type { ReactNode } from 'react';
+import { useEffect, useState } from 'react';
+import { useConnectWallet } from '../hooks/wallet/useConnectWallet.js';
+import { useWalletContext } from '../hooks/wallet/useWalletContext.js';
+import type { StorageAdapter } from '../utils/storageAdapters.js';
+import type { WalletAccount, WalletWithRequiredFeatures } from '@mysten/wallet-standard';
+
+type AutoConnectWalletProps = {
+	storageKey: string;
+	storageAdapter: StorageAdapter;
+	children: ReactNode;
+};
+
+export function AutoConnectWallet({
```

Comment:
Okay @Jordan-Mysten, let me know what you think! This does nicely encapsulate all of the auto-connection / storage-related logic 😄